spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/CronSchedule.java

```diff
@@ -61,17 +61,26 @@
     private String expression;
     private ExecutionTime executionTime;
     private Clock clock;
+    private long scheduleDelay;
 
     public CronSchedule(String expression, ZoneId timezone) {
         this.expression = expression;
         this.timezone = timezone;
         this.executionTime = ExecutionTime.forCron(cronParser.parse(this.expression));
         clock = Clock.system(timezone);
+        this.scheduleDelay = 0;
+    }
+
+    public CronSchedule(String expression, ZoneId timezone, long scheduleDelay) {
+        this(expression, timezone);
+        this.scheduleDelay = scheduleDelay;
     }
 
     public CronSchedule(StreamInput input) throws IOException {
         timezone = input.readZoneId();
         expression = input.readString();
+        Long delayIn = input.readOptionalLong();
+        scheduleDelay = delayIn == null ? 0 : delayIn;
```

Comment:
same comment as in the `writeTo` if we write to stream as non optional we don't need to do this, and since this is always present we can write as non optional in `writeTo` and read it as non optional.